baseplate/lib/thrift_pool.py

```diff
@@ -152,7 +153,7 @@ def __init__(
         endpoint: config.EndpointConfiguration,
         size: int = 10,
         max_age: int = 120,
-        timeout: int = 1,
+        timeout: Union[int, float] = 1,
```

Comment:
nit: I think `float` is sufficient here and you don't actually need to union these.

```python
def foo(x: float) -> None:
    print(x)

foo(1)
foo(1.0)
```

```console
$ mypy foo.py
Success: no issues found in 1 source file
```